resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -1372,6 +1372,10 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
                         qCh, queryId, res.sender());
                 res.release();
                 return;
+            } else if (qCtx.isDone()) {
+                logger.debug("{} Received a DNS response for a query that was completed already ID: UDP [{}: {}]",
```

Comment:
"was completed already" may create an impression that it successfully completed. Since we know it happens only for timeout (or cancellation if we address previous comment), will be helpful to clarify these conditions in the log message.